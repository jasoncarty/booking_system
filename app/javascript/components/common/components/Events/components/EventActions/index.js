import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import DeleteEventButton from './../DeleteEventButton';
import CancelEventButton from './../CancelEventButton';
import JoinEventButton from './../JoinEventButton';

class EventActions extends Component {
  static propTypes = {
    adminSection: PropTypes.bool,
    onDeleteSuccess: PropTypes.func.isRequired,
    onDeleteError: PropTypes.func.isRequired,
    onCancelError: PropTypes.func.isRequired,
    onUpdateAttendance: PropTypes.func.isRequired,
    onJoinError: PropTypes.func.isRequired,
    attendances: PropTypes.arrayOf(PropTypes.number),
    id: PropTypes.number.isRequired
  }

  render() {
    const {
      adminSection,
      onDeleteSuccess,
      onDeleteError,
      id
    } = this.props;

    if (adminSection) {
      return (
        <Fragment>
          <DeleteEventButton
            eventId={id}
            onSuccess={onDeleteSuccess}
            onError={onDeleteError}
          />
          <div className="go-to-event-button">
            <a href={`/admin/events/${id}/edit`}>
              <i className='glyphicon glyphicon-chevron-right'></i>
            </a>
          </div>
        </Fragment>
      );
    }

    const {
      attendances,
      onUpdateAttendance,
      onCancelError,
      onJoinError
    } = this.props;

    return (
      <div className="row spacer-25 actions">
        <div className="col-sm-12">
          {attendances && attendances.includes(id) ? (
            <CancelEventButton
              onSuccess={onUpdateAttendance}
              onError={onCancelError}
              id={id}
            />
          ) : (
            <JoinEventButton
              onSuccess={onUpdateAttendance}
              onError={onJoinError}
              id={id}
            />
          )}
        </div>
      </div>
    );
  }
}

export default EventActions;
