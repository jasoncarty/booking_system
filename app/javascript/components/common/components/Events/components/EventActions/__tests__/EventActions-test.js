import React from 'react';
import { create } from 'react-test-renderer';

import EventActions from './../index';

jest
  .mock('./../../DeleteEventButton', () => 'DeleteEventButton')
  .mock('./../../CancelEventButton', () => 'CancelEventButton')
  .mock('./../../JoinEventButton', () => 'JoinEventButton');

const defaultProps = {
  adminSection: true,
  onDeleteSuccess: () => {},
  onDeleteError: () => {},
  onCancelError: () => {},
  onUpdateAttendance: () => {},
  onJoinError: () => {},
  attendances: [1, 2, 3],
  id: 1
};

describe('<EventActions />', () => {
  it('matches snapshot when admin section', () => {
    const element = create(<EventActions {...defaultProps} />);
    expect(element).toMatchSnapshot();
  });

  it('matches snapshot when not admin section', () => {
    const props = {
      ...defaultProps,
      adminSection: false
    };
    const element = create(<EventActions {...props} />);
    expect(element).toMatchSnapshot();
  });
});
