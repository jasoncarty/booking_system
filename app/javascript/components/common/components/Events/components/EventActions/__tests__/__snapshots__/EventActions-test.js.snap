// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventActions /> matches snapshot when admin section 1`] = `
Array [
  <DeleteEventButton
    eventId={1}
    onError={[Function]}
    onSuccess={[Function]}
  />,
  <div
    className="go-to-event-button"
  >
    <a
      href="/admin/events/1/edit"
    >
      <i
        className="glyphicon glyphicon-chevron-right"
      />
    </a>
  </div>,
]
`;

exports[`<EventActions /> matches snapshot when not admin section 1`] = `
<div
  className="row spacer-25 actions"
>
  <div
    className="col-sm-12"
  >
    <CancelEventButton
      id={1}
      onError={[Function]}
      onSuccess={[Function]}
    />
  </div>
</div>
`;
